src/core/Do.Architecture/Architecture/Application.cs

```diff
@@ -20,6 +23,36 @@ internal Application With(ApplicationDescriptor descriptor)
         return this;
     }
 
+    void CheckDuplicates(ApplicationDescriptor descriptor)
+    {
+        try
+        {
+            descriptor.Layers.ToDictionary(
+                keySelector: layer => layer.Id,
+                elementSelector: layer => layer
+            );
+            descriptor.Features.ToDictionary(
+                keySelector: feature => feature.ToString() ?? feature.GetType().Name,
+                elementSelector: feature => feature
+            );
```

Comment:
to try to build a dictionary to validate if layer id is distinct is kind of a hacking. you're trying to build a dictionary that you'll never use. create a better way to make the validation